storage/src/journal/fixed.rs

```diff
@@ -381,79 +384,70 @@ impl<E: Storage + Metrics, A: Codec<Cfg = ()> + FixedSize> Journal<E, A> {
         Ok(A::decode(&buf[..A::SIZE]).unwrap())
     }
 
-    /// Returns an unordered stream of all items in the journal with position >= `start_pos`.
+    /// Returns an ordered stream of all items in the journal with position >= `start_pos`.
     ///
     /// # Integrity
     ///
     /// If any corrupted data is found, the stream will return an error.
-    ///
-    /// # Concurrency
-    ///
-    /// The `concurrency` parameter controls how many blobs are replayed concurrently. This can
-    /// dramatically speed up the replay process if the underlying storage supports concurrent reads
-    /// across different blobs.
     pub async fn replay(
         &self,
-        concurrency: usize,
         buffer: usize,
         start_pos: u64,
     ) -> Result<impl Stream<Item = Result<(u64, A), Error>> + '_, Error> {
-        assert!(concurrency > 0);
         // Collect all blobs to replay
         let start_blob = start_pos / self.cfg.items_per_blob;
-        let blobs = self
-            .blobs
-            .range(start_blob..)
-            .map(|(index, blob)| (index, blob.clone()))
-            .collect::<Vec<_>>();
+        let blobs = self.blobs.range(start_blob..).collect::<Vec<_>>();
+
+        // Need to gather the sizes outside the closure since flat_map doesn't allow async.
+        let mut sizes = Vec::with_capacity(blobs.len());
+        for b in blobs.iter() {
+            sizes.push(b.1.size().await);
+        }
+        // Create an iterator of (iteration, blob_index, blob, size) tuples to stream.
+        let blob_plus = blobs
+            .into_iter()
+            .zip(sizes.into_iter())
+            .enumerate()
+            .map(|(i, ((blob_index, blob), sz))| (i, *blob_index, (*blob).clone(), sz));
 
-        // Replay all blobs concurrently and stream items as they are read (to avoid occupying too
-        // much memory with buffered data)
         let items_per_blob = self.cfg.items_per_blob;
```

Comment:
nit: prefer grouping this with the stream block before and retainign some comment.

Reply:
done (didn't actually mean to delete that comment!)